### PR TITLE
Feature lightmap radial

### PR DESCRIPTION
**Pull request by Marble879**
### Changes

1. Added a lightmap module that keeps track of specific sensors and their respective lightmap visualization.
2. Added functions to convert sensor readings to appropriate gradients to map the different visualizations based on light values
3. Added unit tests to test the main functionality.

### Ticket resolved
https://trello.com/c/leOTTgbU

### CODE REVIEW

**Review by ChrisAxt — APPROVED** [review]
Good job! Really nice and neat! Nice that tests and documentation was included!

File: homeassistant/components/lightmap/lightmap.py [hunk 1]
```diff
@@ -2,4 +2,248 @@
...
+        root = tree.getroot()
+        root.append(radial_gradient)
+        tree.write(self._svg_path)
+
+    def calculate_light_radial(self):
+        """
+        Calculate light radial distance.
+
+        Assumption: We assume that photoresistors are used where
+        the light intesnity is proportional to the resistance -
```
Comment (AndreaBForce) [inline homeassistant/components/lightmap/lightmap.py]: typo

File: homeassistant/components/lightmap/lightmap.py [hunk 2]
```diff
@@ -2,4 +2,248 @@
...
+        in other words, as light intensity increase, resistance increases.
```
Comment (AndreaBForce) [inline homeassistant/components/lightmap/lightmap.py]: Yes but, in practice the photoresistor when the luminosity is higher the resistance is lower, and when the luminosity is low the resistance is higher. But yeah the value that we read is the luminosity that we read, high value, high luminosity low value, low luminosity.

It was just a technicality in the description, the code is great.
https://en.wikipedia.org/wiki/Photoresistor
https://www.sciencedirect.com/topics/engineering/photoresistors

Comment (AndreaBForce) [reply homeassistant/components/lightmap/lightmap.py]: So I just don't understand this step here, we are reading high value when high luminosity, and low value when low luminosity right?

**Review by AndreaBForce — CHANGES_REQUESTED** [review]
All good, nice and clean and great tests and documentations,
I just have a question about the readings of the photoresistor, but other than that is all good!
Good job

File: homeassistant/components/lightmap/lightmap.py (hunk 2 above)
Comment (AndreaBForce) [reply homeassistant/components/lightmap/lightmap.py]: All ist gud

**Review by Sara-Maha — APPROVED** [review]
Nice job!

**Review by pamali-gu — APPROVED** [review]
Nicely done

**Review by AndreaBForce — APPROVED** [review]
approved

**Pull request merged**